### PR TITLE
fix: handling varfiles in remote actions

### LINKED ISSUE #6146: Varfiles with "type: helm" and sources.repository.url not working
### Current behavior

env file not found (is looking in wrong directory for the varfiles in .garden/sources/…)
```sh
Error processing config for Deploy action redis:
Could not find varfile at path './.env'. Absolute path: /…/garden/examples/redis/.garden/sources/action/deploy.redis--88237447a8/.env
```

### Expected behavior

varfiles should be searched within the action directory

### Reproducible example

garden.yml
```yaml
apiVersion: garden.io/v1
kind: Project
name: redis
environments:
  - name: local
    variables:
      baseHostname: redis.local.demo.garden
providers:
  - name: local-kubernetes
    environments: [local]
variables:
  userId: ${kebabCase(local.username)}

```
redis.garden.yml
```yaml
kind: Deploy
description: Redis service for queueing votes before they are aggregated
type: helm
name: redis
varfiles:
  - ./.env
source:
  repository: 
    url: https://github.com/bitnami/charts#main
spec:
  chart:
    path: bitnami/redis
  values:
    auth:
      enabled: false

```
.env
```
TEST=test
```

```sh
garden dev
garden deploy
```

### Workaround

don't use varfiles

### Suggested solution(s)

varfiles location should be relative to the action yaml definition file

### Your environment

* **OS:** macOS
* **How I'm running Kubernetes:** docker desktop

`0.13.30`

### PR DESCRIPTION
**Pull request by ManAnRuck**
**What this PR does / why we need it**:

fix read varfiles in helm deployments with `sources.repository.url`

**Which issue(s) this PR fixes**:

Fixes #6146

**Special notes for your reviewer**:
It's just a dirty try to fix it. maybe there are some better solutions

### CODE REVIEW

**Comment by vvagaytsev** [comment]
@ManAnRuck thank you so much for creating the PR! 👍 

I'll take a closer look soon. Let's bring this together to the next release :)

**Comment by ManAnRuck** [comment]
i am currently thinking about what is a better solution.

- trying to resolve varfiles and value files relative to original file
- or copy varfiles and valuefiles into the .garden action directory
any suggestions or other ideas?

**Comment by vvagaytsev** [comment]
@ManAnRuck thanks for the fix! LGTM 👍 

@stefreak @thsig can you take a look please? I've added a few more changes to the PR.

**Review by stefreak — APPROVED** [review]
Looks good to me 👍

**Comment by ManAnRuck** [comment]
Thank you :) then I will try now to reuse this function to get the valueFiles too 😃

File: examples/remote-sources/worker/garden.yml [hunk 1]
```diff
@@ -5,6 +5,10 @@ source:
   repository:
     url: https://github.com/garden-io/garden-example-remote-module-jworker.git#main
 
+# Do not delete, this must be resolved from the current path, not from the cloned sourced path
+varfiles:
+  - ./.env
```
Comment (stefreak) [inline examples/remote-sources/worker/garden.yml:10]: @vvagaytsev I just noticed this is an example, and not a test case– examples should be concise and this comment might confuse new users. Maybe it's better to create a more specific integ or unit test for this behaviour and remove this code from the example.

**Pull request merged**